partiql-parser/src/lexer/mod.rs

```diff
@@ -450,27 +469,8 @@ mod tests {
         let query = r#" ` "fooo` "#;
         let mut offset_tracker = LineOffsetTracker::default();
         let toks: Result<Vec<_>, _> = PartiqlLexer::new(query, &mut offset_tracker).collect();
-        assert!(toks.is_err());
-        let error = toks.unwrap_err();
-
-        assert!(matches!(
-            error,
-            ParseError::LexicalError(Located {
-                inner: LexError::UnterminatedIonLiteral,
-                location: Location {
-                    start: BytePosition(ByteOffset(1)),
-                    end: BytePosition(ByteOffset(10))
-                }
-            })
-        ));
-        assert_eq!(
-            error.to_string(),
-            "Lexing error: unterminated ion literal at `(b1..b10)`"
-        );
-        assert_eq!(
-            LineAndColumn::from(offset_tracker.at(query, BytePosition::from(1)).unwrap()),
-            LineAndColumn::new(1, 2).unwrap()
-        );
+        // ion is not eagerly parsed, so unterminated ion does not cause a lex/parse error
+        assert!(toks.is_ok());
```

Comment:
nit: could also rename these tests beginning w/ `err_...` that now don't throw any lexing/parsing error